src/clr-angular/utils/drag-and-drop/draggable.spec.ts

```diff
@@ -6,32 +6,83 @@
-                TestBed.configureTestingModule({
-                    declarations: [NestedHandleTest, ClrDraggable, ClrDragHandle],
-                    providers: [ClrDragAndDropEventBus]
+                TestBed.configureTestingModule(
+                    {imports: [ClrDragAndDropModule, NoopAnimationsModule], declarations: [BasicDraggableTest]});
+                TestBed.overrideComponent(ClrDraggable, {
+                    set: {
+                        providers: [
+                            DomAdapter, ClrDragHandleRegistrar, ClrDraggableSnapshot, ClrGlobalDragMode,
+                            MOCK_DRAG_EVENT_LISTENER_PROVIDER
+                        ]
+                    }
                 });
 
-                this.fixture = TestBed.createComponent(NestedHandleTest);
+                this.fixture = TestBed.createComponent(BasicDraggableTest);
                 this.testComponent = this.fixture.componentInstance;
                 this.draggable = this.fixture.debugElement.query(By.directive(ClrDraggable));
                 this.dragEventListener = this.draggable.injector.get(ClrDragEventListener);
                 this.dragHandleRegistrar = this.draggable.injector.get(ClrDragHandleRegistrar);
+                this.globalDragMode = this.draggable.injector.get(ClrGlobalDragMode);
+                this.fixture.detectChanges();
             });
 
             afterEach(function() {
                 this.fixture.destroy();
             });
 
+            it("should have draggable class", function() {
+                expect(this.draggable.nativeElement.classList.contains("draggable")).toBeTruthy();
+            });
+
+            it("should emit event on drag start and have being-dragged class", function() {
+                spyOn(this.globalDragMode, "enter");
+                const mockDragStartEvent = {dragPosition: {pageX: 11, pageY: 22}};
+                this.dragEventListener.dragStarted.next(mockDragStartEvent);
+                this.fixture.detectChanges();
+                expect(this.testComponent.dragStartEvent).toBe(mockDragStartEvent);
+                expect(this.testComponent.dragMoveEvent).toBeUndefined();
+                expect(this.testComponent.dragEndEvent).toBeUndefined();
+                expect(this.draggable.nativeElement.classList.contains("being-dragged")).toBeTruthy();
+                expect(this.globalDragMode.enter).toHaveBeenCalled();
+            });
+
+            it("should emit event on drag move", function() {
+                const mockDragMoveEvent = {dragPosition: {pageX: 33, pageY: 44}};
+                this.dragEventListener.dragMoved.next(mockDragMoveEvent);
+                expect(this.testComponent.dragStartEvent).toBeUndefined();
+                expect(this.testComponent.dragMoveEvent).toBe(mockDragMoveEvent);
+                expect(this.testComponent.dragEndEvent).toBeUndefined();
+            });
+
+            it("should emit event on drag end", function() {
```

Comment:
Here the description is missing the class test that you are doing. If you are deciding to break the earlier test into 2 parts then this should be too. If not, the description should be updated to include the class details.

Reply:
Thanks. Addressing this right now.